torch/package/package_importer.py

```diff
@@ -565,11 +586,11 @@ def _add_file(self, filename: str):
         if len(prefix) > 1 and prefix[0] == ".data":
             return
         package = self._get_or_create_package(prefix)
-        if isinstance(package, _ExternNode):
-            raise ImportError(
-                f"inconsistent module structure. package contains a module file {filename}"
-                f" that is a subpackage of a module marked external."
-            )
+        # if isinstance(package, _ExternNode):
```

Comment:
We probably want to tighten check instead of throwing it out completely, e.g. we distinguish `ExternNode` from `SelectiveExtern` node, and still perform the check for regular `ExternNode`s